Google analytics

Emmanueldmlr opened the pull request:
# Google Analytics Integration

_Integration of google analytics for page views and tracking actions

## How should this be tested?

1. Log on to the google analytics dashboard to check real-time reporting

![image](https://user-images.githubusercontent.com/70170061/164441617-13d20327-5a46-4cd7-9baa-0374b57bcfcd.png)


## Notes or observations
This PR also fixes the issue with the analytics image click

## Linked issues

fixes https://github.com/EveripediaNetwork/issues/issues/267
fixes https://github.com/EveripediaNetwork/issues/issues/275

kesar commented:
it looks promising make sure it logs properly the routes / pageviews ( https://mariestarck.com/add-google-analytics-to-your-next-js-application-in-5-easy-steps/ ) 

we will also want to track events, a first event you can track for testing its each time that people click in wallet drawer or when people submit a wiki

Emmanueldmlr commented:
Okay then. Thanks for the resource, I will check it out.

Emmanueldmlr commented:
![image](https://user-images.githubusercontent.com/70170061/164682893-cfef7a40-b486-46a1-8f3c-03e13a930895.png)
@kesar, I implemented page view tracking and also added "SUBMIT_WIKI" event, which gets fired when the publish button is clicked.

Emmanueldmlr commented:
I think it's easy to iterate on it now.

kesar commented on src/pages/create-wiki/index.tsx:
```diff
@@ -231,6 +232,11 @@ const CreateWiki = () => {
 
+    logEvent({
+      action: 'SUBMIT_WIKI',
+      params: { address: accountData?.address, title: wiki.title },
```
use slug since its the wiki identifier

Emmanueldmlr replied:
Okay then

kesar commented on src/utils/googleAnalytics.ts:
```diff
@@ -0,0 +1,20 @@
+export const pageView = (url: URL) => {
+  const GA_TRACKING_ID = process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS as string
+  if (typeof window.gtag !== 'undefined') {
+    window.gtag('config', GA_TRACKING_ID, {
```
i think this config its to set the first time configuration. if you need to send page view you should use:

gtag('event', 'page_view', {
  page_title: '<Page Title>',
  page_location: '<Page Location>',
  page_path: '<Page Path>',
  send_to: '<GA_MEASUREMENT_ID>'
})

https://developers.google.com/analytics/devguides/collection/gtagjs/pages

Emmanueldmlr replied:
I think it's just a manual way of sending page views. meanwhile, page_title, page_location can also be added to the pageView parameters and we have the details on the analytics page.

Emmanueldmlr replied:
![Uploading image.png…]()

Emmanueldmlr replied:
![image](https://user-images.githubusercontent.com/70170061/164689580-34fa1006-e99a-4512-836b-0dee79217e99.png)

kesar replied:
you are calling a config command each time there is a new view and that's wrong (it can work now, but it could stop working anytime).

config its to setup a configuration once and leave it like that. You already have it here: https://github.com/EveripediaNetwork/ep-ui/pull/186/files#diff-88a36bc7a53bfa58c6f451464798d631d785160dfacfdaf9479be7a0b1d7e5e5R64

If that works fine (with paths & titles) then you dont need to manually send the pageviews,
if that does not work fine (no titles) then you need to disable pageviews in config and trigger the manually event in the pageView function you created

Emmanueldmlr replied:
Oh. I understand what you meant now.

kesar commented on src/pages/_app.tsx:
```diff
@@ -37,6 +51,22 @@ const App = (props: EpAppProps) => {
+      <Script
+        id="google-analytics"
+        strategy="lazyOnload"
+        src={`https://www.googletagmanager.com/gtag/js?id=${process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS}`}
```
we forgot to add this to .env.sample :/

Emmanueldmlr replied:
thats true. But I created the variable on Vercel already. I think you just have to update it to Everipedia credentials. and I will just add it to .env.sample for my new update.
